### ISSUE TITLE
Help with gitflow, commitizen and automatic version bump

### ISSUE THREAD

**Issue by krish2487** [commitizen-tools/commitizen#358]
## Description
Hello folks,

I apologize if this is a silly issue to ask, but I need some help with git flow and commitizen - specifically for the release flow.

I use commitizen for standardizing the commit messages presently. I see that commitizen also allows for the autobump of versions and changelog generation based on the commits. I would like to use this in conjunction with git flow. Gitflow does not have a mechanism for automatically bumping the release versions, commitizen does. commitizen does not manage feature branches, release branches and merges, gitflow does.

In short, I would like to use gitflow to manage branches and features and I would like to use commitizen to manage commit styles and automatic version bumps.

## Possible Solution

This is more or less what I have in mind.

```
git flow init -d

cz init

git flow feature start <feature_name>
__commits using `cz commit`__
git flow feature finish <feature_name>

cz bump --changelog --yes --dry-run

git flow release start (output from the above command)
cz bump --changelog --yes
git flow release finish <release version>

```

## Additional context

There is a small problem with the above workflow. 
The moment I run the `cz bump` command without the --dry-run option in the release section. It creates the changelog and also creates a tag.
That prevents the git flow release finish command to error and not complete.
Specifically this message
```
Fatal: Tag already exists and does not point to release branch
```

## Related Issue
I have also created an [issue](https://github.com/petervanderdoes/gitflow-avh/issues/451) on gitflow repo since this request affects both projects.

I was wondering if anyone else has a solution to the above, rather silly issue. 
In short, I am looking for a solution to automatically incrementing the semver of the code, and changelog generation. 
I will be very grateful if anyone can help with a workaround or provide with an alternative to the automatic version increment. 

Thank you in advance!! :-)

PS : This is most likely not a feature request.. but it certainly not a documentation issue or a bug. It is most likely an issue with my understanding on the workflow.. but a feature request is the closest description that suits the issue.

**Comment by Lee-W** [comment]
If my understanding is correct, the main problem is this part. The command `git flow release finish <release version>` will try to create a git tag that has already been created by commitizen.

```sh
git flow release start (output from the above command)
cz bump --changelog --yes
git flow release finish <release version>
```

Do you think something like `git flow release finish -n <release version>` could solve your problem?
https://stackoverflow.com/questions/33734731/stop-git-flow-from-creating-a-tag-automatically

**Comment by krish2487** [comment]
@Lee-W  
Thank you for the prompt response.
Yes. It does look like that the correct, appropriate solution. 
Though.. I m not sure if the tags would be preserved once the release is finished. 
Let me experiment and get back with findings. :-)

Thank you again!! This is very helpful!

**Comment by krish2487** [comment]
Well it looks like you hit the nail on the head @Lee-W . :-D 
I created a test repo and made a series os trivial commits to test the suggestion.

Here is what I ended up with.

<img width="772" alt="Screen Shot 2021-03-02 at 11 17 05" src="https://user-images.githubusercontent.com/23363406/109633946-fbf60c00-7b48-11eb-9b0a-32d448a5e83d.png">

The tags are not conflicting. and the tags are preserved in the release branch history. 

Thank you again for the extremely prompt response and correct solution! :-D.

I am closing the issue now.
